cpp/zotaut/journal_timeliness_checker.cc

```diff
@@ -36,22 +36,33 @@ namespace {
 
 
 [[noreturn]] void Usage() {
-    ::Usage("[--min-log-level=log_level] [--default-update-window=no_of_days] config_file_path sender_email_address "
-            "notification_email_address");
+    ::Usage("[--min-log-level=log_level] [--default-update-window=no_of_days] sender_email_address notification_email_address");
+}
+
+
+bool MaxDeliveredAtSmallerThanUpdateWindow(const time_t max_delivered_at, const unsigned update_window) {
+     if (max_delivered_at == TimeUtil::BAD_TIME_T)
+         return false;
+     return max_delivered_at < ::time(nullptr) - update_window * 86400;
 }
 
 
 void ProcessJournal(ZoteroHarvester::Util::UploadTracker * const upload_tracker, const std::string &journal_name,
                     const std::string &zeder_id, const std::string &zeder_instance, const unsigned update_window,
                     std::string * tardy_list)
 {
-    const time_t max_delivered_at(upload_tracker->getLastUploadTime(StringUtil::ToUnsigned(zeder_id),
+    const time_t all_max_delivered_at(upload_tracker->getLastUploadTime(StringUtil::ToUnsigned(zeder_id),
```

Comment:
Under normal circumstances, we expect the journal to be updated within the update window. This would also mean that the online_first articles would never be removed. Shouldn't we compare each record's timestamp to a max timestamp instead?

Reply:
Addressed in the current code.